client/flutter/lib/api/content_loading.dart

```diff
@@ -35,6 +36,18 @@ class ContentLoading {
         return await _loadFromNetwork(name, languageAndCountry);
       } catch (err) {
         print("Network bundle for $languageAndCountry not found: $err");
+        if (err is ContentBundleVersionException) {
+          // Defer showing the dialog briefly until after screen build.
+          Future.delayed(const Duration(seconds: 1), () {
+            // TODO: Localize
+            Dialogs.showAppDialog(
```

Comment:
Should logic like this go in the widgets instead of in `lib/api`?
You could catch the exception there right?  And avoid the 1 second delay?  Seems fragile this way.

Reply:
You are correct to be skeptical - here are some issues: 1) It's not really the timing here as just deferring the show safely to the UI thread - a delay of zero would be equivalent.   2) We don't want to throw an exception here because we want to proceed and load the local content and just "warn" about it.  We can easily refactor this to an API that would support passing that information back, this was just the quickest / smallest change I could make this morning. 3) I don't like having the BuildContext in this class at all but it is necessary for access to the Locale information (as far as I understand) and therefore we are already sort of bound to the UI layer.  

The combo of the above lead me to thinking this was the best approach for the moment.  I will look for a better way to refactor this today if time permits.

Reply:
I added a TODO for the larger refactoring.